CONTRIBUTING.md

```diff
@@ -73,6 +73,37 @@ The source must be formatted according to the style guide below.
 
 Header includes must be [levelized](./Builds/levelization).
 
+Changes should be usually squashed down into one commit with a
+[good commit message](#good-commit-messages).
+In some cases, it is appropriate to organize the changes into
+multiple commits. In that situation, all of the commits should be
+distinct changes and have
+[good commit messages](#good-commit-messages).
```

Comment:
A non-trivial PR will have more than one commit for a good reason, so I wouldn't say "should be usually squashed into one commit"; instead I suggest something like this:

The purpose of separation of a change into commits is not to reflect the history of how the author implemented the change "warts and all", instead it is to describe the change in smaller (yet well-formed) chunks, that make most sense to the reader. For this reason:

* changes should be presented in a logical sequence of a small number (ideally one, for a small change) of commits
* each commit should have a [good message](#good-commit-messages) to explain a specific aspects of the change.
* every commit should be signed
* every commit should be well-formed (build passing, unit tests passing), as this helps to resolve merge conflicts.

Reply:
I agree that there could be good reasons for multiple commits, but I think that should be an uncommon exception. Most PRs should be opened with a single squashed commit.

I have rewritten this part to reflect the rest of your suggestions.